x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/SecurityServerTransportInterceptor.java

```diff
@@ -155,6 +211,12 @@ private <T extends TransportResponse> void sendWithUser(
             throw new IllegalStateException("there should always be a user when sending a message for action [" + action + "]");
         }
 
+        if (securityContext.getParentAuthorization() != null) {
+            if (remoteClusterAliasResolver.apply(connection).isPresent()) {
+                assert false : "parent authorization header should not be set for remote cluster requests";
+            }
+        }
```

Comment:
nit: I find this distracting, please remove or replace with:
```
assert securityContext.getParentAuthorization() == null || remoteClusterAliasResolver.apply(connection).isPresent() == false : "parent authorization header should not be set for remote cluster requests"
```

Reply:
Replaced https://github.com/elastic/elasticsearch/pull/91886/commits/91f5f4a68b1510c007d32d488d24c5e6634ca620